vllm/forward_context.py

```diff
@@ -21,9 +24,10 @@
 
 @dataclass
 class ForwardContext:
-    static_forward_context: Dict[str, Any]
+    # copy from vllm_config.compilation_config.static_forward_context
+    layers: Dict[str, Any]
     # TODO: extend to support per-layer dynamic forward context
-    dynamic_forward_context: Any
+    attn_metadata: "AttentionMetadata"  # set dynamically for each forward pass
```

Comment:
other changes look good to me, but why do we need to rename these two variables?

Reply:
I think it makes the meaning of the two context more clear. Seems that we only need the attention cls as static ctx and attn_metadata as dynamic ctx.

Reply:
Are you planing to use the ctx to save other information?

Reply:
maybe rename `layers` to `stateful_layers` ? `layers` are too general and confusing

Reply:
Prefer `attn_layers`.  The problem of `stateful_layers` is that given that name, we shouldn't save attention layers that do not need kv cache (e.g., with `AttentionType=ENCODER`) in the dict, and then `unified_attention` cannot be implemented.

Reply:
`attn_layers` makes sense.

Reply:
done